code/datums/datumvars.dm

```diff
@@ -386,7 +386,10 @@ body
 				html += "<ul>"
 				var/index = 1
 				for(var/entry in L)
-					html += debug_variable(index, L[index], level + 1)
+					if(istext(entry))

```

Comment:
Text is still not the only possible key type.

Try `!isnum(entry) && !isnull(L[entry])`